pkg/proxydisc/client.go

```diff
@@ -0,0 +1,199 @@
+package proxydisc
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/SkycoinProject/dmsg/cipher"
+	"github.com/sirupsen/logrus"
+
+	"github.com/SkycoinProject/skywire-mainnet/internal/httpauth"
+)
+
+// Config configures the HTTPClient.
+type Config struct {
+	PK       cipher.PubKey
+	SK       cipher.SecKey
+	Port     uint16
+	DiscAddr string
+}
+
+// HTTPClient is responsible for interacting with the proxy-discovery
+type HTTPClient struct {
+	log    logrus.FieldLogger
+	conf   Config
+	entry  Proxy
+	auth   *httpauth.Client
+	client http.Client
+}
+
+// NewClient creates a new HTTPClient.
+func NewClient(log logrus.FieldLogger, conf Config) *HTTPClient {
+	return &HTTPClient{
+		log:  log,
+		conf: conf,
+		entry: Proxy{
+			Addr: NewSWAddr(conf.PK, conf.Port),
+		},
+		client: http.Client{},
+	}
+}
+
+func (c *HTTPClient) addr(path string) string {
+	return c.conf.DiscAddr + path
+}
+
+// Auth returns the internal httpauth.Client
+func (c *HTTPClient) Auth(ctx context.Context) (*httpauth.Client, error) {
+	if c.auth != nil {
+		return c.auth, nil
+	}
+	fmt.Println(c.conf.DiscAddr)
```

Comment:
Is it intended ot was left after debugging?

Reply:
Left after debugging. Thanks for the heads up.